### PR TITLE
[OC-1434] - Rework Edit/Create/Delete modals (Admin Screen)

### PR DESCRIPTION
**Pull request by AlexGuironnetRTE**
In addition to replacing the Angular Material inputs by our input components, I changed the Entity/Groups dropdowns so they display names rather than ids.
To do so, I had to rely on an initial load of the data at startup that was already existing for entities but that I had to create for groups. I chose to do the same as for entities, which meant allowing GET for all users (not just ADMIN users), to keep it coherent and simple, but if groups data is more sensitive we can also change the code in the front so the initialization is only done for ADMIN users.

In addition, now the the Entity/Groups dropdowns display names, I think we should state in the docs (and preferably enforce in the API) the fact that names should be unique too ?

For release notes:
Under tasks:
[OC-1434] - Rework Edit/Create/Delete modals (Admin Screen)
Mention the API changes ?

[OC-1434]: https://opfab.atlassian.net/browse/OC-1434

### CODE REVIEW

**Review by freddidierRTE — COMMENTED** [review]
> In addition to replacing the Angular Material inputs by our input components, I changed the Entity/Groups dropdowns so they display names rather than ids.
> To do so, I had to rely on an initial load of the data at startup that was already existing for entities but that I had to create for groups. I chose to do the same as for entities, which meant allowing GET for all users (not just ADMIN users), to keep it coherent and simple, but if groups data is more sensitive we can also change the code in the front so the initialization is only done for ADMIN users.
> 
--> Information is not sensitive , but i would prefer initialize the list in the admin component as it is only use for this screen , for the majority of users , there will download this list for nothing.

> In addition, now the the Entity/Groups dropdowns display names, I think we should state in the docs (and preferably enforce in the API) the fact that names should be unique too ?

--> To add as a new issue in Jira 

> 
> For release notes:
> Under tasks:
> [OC-1434](https://opfab.atlassian.net/browse/OC-1434) - Rework Edit/Create/Delete modals (Admin Screen)
> Mention the API changes ?

--> No

File: ui/main/src/app/modules/admin/components/editmodal/users/edit-user-modal.component.ts [hunk 1]
```diff
@@ -44,23 +49,64 @@ export class EditUserModalComponent implements OnInit {
...
+    });
+
 
   }
 
-  update() {
+  ngOnDestroy() {
+    this.unsubscribe$.next();
+    this.unsubscribe$.complete();
+  }
```
[the remark this answers is not in the document]
Comment (AlexGuironnetRTE) [reply ui/main/src/app/modules/admin/components/editmodal/users/edit-user-modal.component.ts:103]: Correct, I removed it.

File: ui/main/src/app/app.component.ts [hunk 2]
```diff
@@ -136,6 +137,7 @@ export class AppComponent implements OnInit {
           merge(
             this.userService.loadUserWithPerimetersData(),
             this.entitiesService.loadAllEntitiesData(),
+            this.groupsService.loadAllGroupsData(),
```
[the remark this answers is not in the document]
Comment (AlexGuironnetRTE) [reply ui/main/src/app/app.component.ts]: In the end I moved the loading of group data to after the "main" init has taken place so I don't need to change it.

By the way, isn't there a (very small) risk of not waiting for one of the steps to emit if another one emits twice before (namely initCardSubscription as the others are http requests that will only emit once by design)?

File: services/core/users/src/main/java/org/lfenergy/operatorfabric/users/configuration/oauth2/WebSecurityConfiguration.java [hunk 3]
```diff
@@ -75,6 +75,7 @@ public static void configureCommon(final HttpSecurity http) throws Exception {
                 .antMatchers(HttpMethod.PATCH, USERS_SETTINGS_PATH).access(IS_ADMIN_OR_OWNER)
                 .antMatchers(HttpMethod.GET, USERS_PERIMETERS_PATH).access(IS_ADMIN_OR_OWNER)
                 .antMatchers(USERS_PATH).hasRole(ADMIN_ROLE)
+                .antMatchers(HttpMethod.GET, GROUPS_PATH).authenticated()
                 .antMatchers(GROUPS_PATH).hasRole(ADMIN_ROLE)
```
[the remark this answers is not in the document]
Comment (AlexGuironnetRTE) [reply services/core/users/src/main/java/org/lfenergy/operatorfabric/users/configuration/oauth2/WebSecurityConfiguration.java]: Taking into account your comment above I'm finally not changing the API security, but the way I understood the above was:
allow only GET methods on groups path for all authenticated users
in addition, allow all methods on groups path for admin users only

So the antMatchers(GROUPS_PATH).hasRole(ADMIN_ROLE) would have been needed so admin users could still update, delete and create groups.

**Comment by AlexGuironnetRTE** [comment]
> > In addition, now the the Entity/Groups dropdowns display names, I think we should state in the docs (and preferably enforce in the API) the fact that names should be unique too ?
> 
> --> To add as a new issue in Jira

Done : https://opfab.atlassian.net/browse/OC-1456

File: ui/main/src/app/app.component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/app.component.ts]: initCardSubscription is not in the merge / skip mechanism

**Pull request merged**